pkg/report/sarif.go

```diff
@@ -0,0 +1,213 @@
+					vuln.VulnerabilityID, vuln.Severity, vuln.PkgName, vuln.FixedVersion, vuln.VulnerabilityID, vuln.PrimaryURL, vuln.Description),
+				helpMarkdown: fmt.Sprintf("**Vulnerability %v**\n| Severity | Package | Fixed Version | Link |\n| --- | --- | --- | --- |\n|%v|%v|%v|[%v](%v)|\n\n%v",
+					vuln.VulnerabilityID, vuln.Severity, vuln.PkgName, vuln.FixedVersion, vuln.VulnerabilityID, vuln.PrimaryURL, vuln.Description),
+				message: fmt.Sprintf("Package: %v\nInstalled Version: %v\nVulnerability %v\nSeverity: %v\nFixed Version: %v\nLink: [%v](%v)",
+					vuln.PkgName, vuln.InstalledVersion, vuln.VulnerabilityID, vuln.Severity, vuln.FixedVersion, vuln.VulnerabilityID, vuln.PrimaryURL),
+			})
+		}
+		for _, misconf := range res.Misconfigurations {
+			sw.addSarifResult(&sarifData{
+				vulnerabilityId:  misconf.ID,
+				severity:         misconf.Severity,
+				url:              misconf.PrimaryURL,
+				resourceClass:    string(res.Class),
+				artifactLocation: toPathUri(res.Target),
+				resultIndex:      getRuleIndex(misconf.ID, ruleIndexes),
+				fullDescription:  html.EscapeString(misconf.Description),
+				helpText: fmt.Sprintf("Misconfiguration %v\nType: %s\nSeverity: %v\nCheck: %v\nMessage: %v\nLink: [%v](%v)\n%s",
+					misconf.ID, misconf.Type, misconf.Severity, misconf.Title, misconf.Message, misconf.ID, misconf.PrimaryURL, misconf.Description),
+				helpMarkdown: fmt.Sprintf("**Misconfiguration %v**\n| Type | Severity | Check | Message | Link |\n| --- | --- | --- | --- | --- |\n|%v|%v|%v|%s|[%v](%v)|\n\n%v",
+					misconf.ID, misconf.Type, misconf.Severity, misconf.Title, misconf.Message, misconf.ID, misconf.PrimaryURL, misconf.Description),
+				message: fmt.Sprintf("Artifact: %v\nType: %v\nVulnerability %v\nSeverity: %v\nMessage: %v\nLink: [%v](%v)",
+					res.Target, res.Type, misconf.ID, misconf.Severity, misconf.Message, misconf.ID, misconf.PrimaryURL),
+			})
+		}
+	}
+	sw.run.ColumnKind = columnKind
+	sw.run.OriginalUriBaseIDs = map[string]*sarif.ArtifactLocation{
+		"ROOTPATH": {URI: &rootPath},
+	}
+	sarifReport.AddRun(sw.run)
+	return sarifReport.PrettyWrite(sw.Output)
+}
+
+func toSarifRuleName(class string) string {
+	switch class {
+	case ClassOSPkg:
+		return sarifOsPackageVulnerability
+	case ClassLangPkg:
+		return sarifLanguageSpecificVulnerability
+	case ClassConfig:
+		return sarifConfigFiles
+	default:
+		return sarifUnknownIssue
+	}
+}
+
+func toSarifErrorLevel(severity string) string {
+	switch severity {
+	case "CRITICAL", "HIGH":
+		return sarifError
+	case "MEDIUM":
+		return sarifWarning
+	case "LOW", "UNKNOWN":
+		return sarifNote
+	default:
+		return sarifNone
+	}
+}
+
+func toPathUri(input string) string {
+	var matches = re.FindStringSubmatch(input)
+	if matches != nil {
+		input = matches[re.SubexpIndex("path")]
+	}
+	return strings.ReplaceAll(input, "\\", "/")
+}
+
+func sarifSeverityLevel(severity string) string {
```

Comment:
Can we use CVSS score?

Reply:
If you could put the CVSS score into a tag, that would make it filterable in the github security dashboard

Reply:
> Can we use CVSS score?

I can't find CVSS score for misconfigurations...

Reply:
Oh, you're right. We need this conversion for misconfigurations. Can we use CVSS only for vulnerabilities?